mvlearn/embed/dcca.py

```diff
@@ -0,0 +1,349 @@
+    >>> dcca = DCCA(input_size1, input_size2, outdim_size, layer_sizes1,
+                    layer_sizes2)
+    >>> outputs = dcca.fit_transform([view1, view2])
+    >>> print(outputs[0].shape)
+    (1000, 2)
+
+    References
+    ----------
+    .. [#1DCCA] Andrew, G., Arora, R., Bilmes, J., & Livescu, K. (2013,
+                February). Deep canonical correlation analysis. In
+                International conference on machine learning (pp. 1247-1255).
+    """
+
+    def __init__(
+            self, input_size1=None, input_size2=None, outdim_size=2,
+            layer_sizes1=None, layer_sizes2=None,
+            use_all_singular_values=False, device=torch.device('cpu'),
+            epoch_num=10, batch_size=800, learning_rate=1e-3, reg_par=1e-5,
+            print_train_log_info=False, threshold=1e-2
+            ):
+
+        super().__init__()
+        # check input_size1/2
+
+        self.input_size1 = input_size1
+        self.input_size2 = input_size2
+        self.outdim_size = outdim_size
+
+        if layer_sizes1 is None:
+            self.layer_sizes1 = [1000, outdim_size]
+        if layer_sizes2 is None:
+            self.layer_sizes2 = [1000, outdim_size]
+
+        self.use_all_singular_values = use_all_singular_values
+        self.device = device
+        self.epoch_num = epoch_num
+        self.batch_size = batch_size
+        self.learning_rate = learning_rate
+        self.reg_par = reg_par
+        self.print_train_log_info = print_train_log_info
+        self.threshold = threshold
+
+        self.deep_model = DeepCCA(layer_sizes1, layer_sizes2, input_size1,
+                                  input_size2, outdim_size,
+                                  use_all_singular_values, device=device)
+        self.linear_cca = linear_cca()
+
+        self.model = nn.DataParallel(self.deep_model)
+        self.model.to(device)
+        self.loss = self.deep_model.loss
+        self.optimizer = torch.optim.RMSprop(self.model.parameters(),
+                                             lr=self.learning_rate,
+                                             weight_decay=reg_par)
+
+    def fit(self, Xs, y=None):
+        r"""
+        Fits the deep networks for each view such that the output of the
+        linear CCA has maximum correlation.
+
+        Parameters
+        ----------
+        Xs : list of array-likes or numpy.ndarray
+             - Xs length: n_views
+             - Xs[i] shape: (n_samples, n_features_i)
+            The data to fit to. Each view will receive its own embedding.
+
+        y : Unused parameter for base class fit_transform compliance
+
+        Returns
+        -------
+        self : returns an instance of self.
+        """
+        Xs = check_Xs(Xs, multiview=True)  # ensure valid input
+        x1 = torch.DoubleTensor(Xs[0])
+        x2 = torch.DoubleTensor(Xs[1])
+        x1.to(self.device)
+        x2.to(self.device)
+
+        data_size = x1.size(0)
+
+        checkpoint = 'checkpoint.model'
```

Comment:
is this what pytorch is saving? maybe set it as an attribute name (doesn't have to be a parameter). That way user knows where/what the checkpoints are being named in case fit() crashes.

Reply:
are checkpoints saved? might be useful. can just make an issue for it tho